src/main/java/karuna/karuna_backend/Errors/DatabaseExceptions/DatabaseErrorResolver.java

```diff
@@ -0,0 +1,56 @@
+package karuna.karuna_backend.Errors.DatabaseExceptions;
+
+import karuna.karuna_backend.Errors.ErrorKeys.DataIntegrityErrorKey;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.stereotype.Component;
+
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.Map;
+
+
+/**
+ * Component responsible for resolving database error codes to defined {@link DataIntegrityErrorKey} values.
+ * This mapping helps in translating specific SQL state codes into more generic error descriptions,
+ * making it easier to handle and respond to database errors consistently across the application.
+ */
+@Component
+class DatabaseErrorResolver {
+    private static final Map<String, DataIntegrityErrorKey> errorMappings = new HashMap<>();
+
+    // Static initializer to populate the errorMappings map with common SQL error codes.
+    static{
+        errorMappings.put("1062", DataIntegrityErrorKey.UNIQUE_CONSTRAINT_VIOLATION); // MySQL unique constraint code
```

Comment:
why u add mysql? we use postgresql

Reply:
So it would be easier to switch to a different database without code refactoring and I believe this switch might happen soon